dashboards-reports/public/components/report_definitions/delivery/delivery.tsx

```diff
@@ -55,25 +66,47 @@ export function ReportDelivery(props: ReportDeliveryProps) {
     editDefinitionId,
     reportDefinitionRequest,
     httpClientProps,
+    showDeliveryChannelError,
+    deliveryChannelError
   } = props;
 
-  const [emailCheckbox, setEmailCheckbox] = useState(false);
+  const [sendNotification, setSendNotification] = useState(false);
+  const [channels, setChannels] = useState([]);
+  const [selectedChannels, setSelectedChannels] = useState([]);
+  const [notificationSubject, setNotificationSubject] = useState('');
+  const [notificationMessage, setNotificationMessage] = useState('');
+  const [selectedTab, setSelectedTab] = React.useState<'write' | 'preview'>(
+    'write'
+  );
+  const [testMessageConfirmation, setTestMessageConfirmation] = useState('');
+
+  const handleSendNotification = (e: { target: { checked: boolean | ((prevState: boolean) => boolean); }; }) => {
+    setSendNotification(e.target.checked);
+    includeDelivery = e.target.checked;
+  }
 
-  const handleEmailCheckbox = (e: {
-    target: { checked: React.SetStateAction<boolean> };
-  }) => {
-    setEmailCheckbox(e.target.checked);
-    if (e.target.checked) {
-      // if checked, set delivery type to email
-      reportDefinitionRequest.delivery.delivery_type =
-        DELIVERY_TYPE_OPTIONS[1].id;
-    } else {
-      // uncheck email checkbox means to use default setting, which is opensearch dashboards user
-      defaultCreateDeliveryParams();
+  const handleSelectedChannels = (e: React.SetStateAction<never[]>) => {
+    setSelectedChannels(e);
+    reportDefinitionRequest.delivery.configIds = [];
+    for (let i = 0; i < e.length; ++i) {
+      reportDefinitionRequest.delivery.configIds.push(e[i].label);
     }
-  };
+  }
 
-  const emailDelivery = emailCheckbox ? <EmailDelivery {...props} /> : null;
+  const handleNotificationSubject = (e: { target: { value: React.SetStateAction<string>; }; }) => {
+    setNotificationSubject(e.target.value);
+    reportDefinitionRequest.delivery.title = e.target.value.toString();
```

Comment:
`value` looks to be string from your type definition, why is `toString` needed?

Reply:
All of the `e` types were auto-filled by TypeScript, will review all and change those that are needed